### PR TITLE
Add instructions for setting up and testing project to README

### PR DESCRIPTION
**Pull request by evenskari**
This includes how to
- setup project/getting started section
- configure PostgreSQL
- run project locally
- deployment to AWS
- load test with Tsung

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,3 +26,147 @@ Our application would allow students to intuitively choose the best schedule for
...
+First install Yarn from [here](https://yarnpkg.com/lang/en/docs/install/)<br/>  
+Find your workspace folder and clone the repository
+```
+git clone https://github.com/scalableinternetservices/ucsb_class_scheduler.git
+```
+
+
+While being in the newly cloned directory, configure and update gems by running
+```
+yarn install
```
Comment (estensen) [inline README.md]: Aren't we using `bundle install` to install gems?

File: README.md [hunk 2]
```diff
@@ -26,3 +26,147 @@ Our application would allow students to intuitively choose the best schedule for
...
+
+With a working database you are now ready to start local testing. Before testing, run 
+```
+yarn install
+```
+Then open two terminal consoles. 
+
+To start the backend server, run the following in the main directory
+```
+rails start
```
Comment (estensen) [inline README.md]: Has to be run on port 3001
`rails start -p 3001`

File: README.md [hunk 3]
```diff
@@ -26,3 +26,147 @@ Our application would allow students to intuitively choose the best schedule for
...
+```
+git checkout <desired_branch>
+```
+
+To deploy the instance, run
+```
+eb create -db.engine postgres -db.i db.t2.micro -db.user u --envvars SECRET_KEY_BASE=<random_secret> --single ucsb-class-scheduler-<your_name>
+```
+* Replace <your_name> with your name  
+* Replace <random_secret>  with a semi-random large string of alphanumeric characters. <br/>
```
Comment (estensen) [inline README.md]: Generate secret with `rake secret`

File: README.md [hunk 4]
```diff
@@ -26,3 +26,147 @@ Our application would allow students to intuitively choose the best schedule for
...
+
+When done testing, terminate the instance by running
+```
+eb terminate
+```
+
+# Load testing with Tsung
+
+By load testing we can detect how the website's key performance metrics, like response times and error rates, are affected by different implementations and strategies. This is done by simulating real users using the site. The tests should include both reads and write operations.  
+<br/>
```
Comment (estensen) [inline README.md]: Why is this here?

Comment (estensen) [reply README.md]: (the br tag)

File: README.md [hunk 5]
```diff
@@ -26,3 +26,147 @@ Our application would allow students to intuitively choose the best schedule for
...
+After the test is done, the log files will only be available while the stack is active.
+
+Copy the log files for later review and comparison by opening another terminal in the directory where the .pem file resides. 
+
+Then run
+```
+rsync -auvLe 'ssh -i ucsb-class-scheduler.pem' ec2-user@xx.xx.xx.xx:tsung_logs .
+```
+The log files will then be copied to this directory.  
+<br/>
```
Comment (estensen) [inline README.md]: Remove br tag

**Comment by dannycho7** [comment]
Can we not add this to the README.md? I think it's better to create a new file called "testing.md"

File: README.md (hunk 1 above)
Comment (evenskari) [reply README.md]: Changed to '''bundle install'''

File: README.md (hunk 2 above)
Comment (evenskari) [reply README.md]: Changed

File: README.md (hunk 3 above)
Comment (evenskari) [reply README.md]: Ok

File: README.md (hunk 4 above)
Comment (evenskari) [reply README.md]: Removed

File: README.md (hunk 5 above)
Comment (evenskari) [reply README.md]: Removed

**Comment by estensen** [comment]
I think installation and db config could be in the README. It doesn't belong in TEST.md

File: TESTING.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
+# Local testing
+
+With a working database you are now ready to start local testing. Before testing, run 
+```
+yarn install
```
Comment (estensen) [inline TESTING.md]: Bundle install?
